js/render/core/renderer.js

```diff
@@ -77,6 +81,148 @@ void main() {
 }
 `;
 
+const VERTEX_SHADER_MULTI_DEPTH_ENTRY = `
+uniform mat4 LEFT_PROJECTION_MATRIX, LEFT_VIEW_MATRIX, RIGHT_PROJECTION_MATRIX, RIGHT_VIEW_MATRIX, MODEL_MATRIX;
+out vec4 vWorldPosition;
+
+const mat4 identity = mat4(1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1);
+
+void main() {
+  vWorldPosition = vertex_main(identity, identity, MODEL_MATRIX);
+  gl_Position = vertex_main(
+    (gl_ViewID_OVR == 0u) ? LEFT_PROJECTION_MATRIX : RIGHT_PROJECTION_MATRIX,
+    (gl_ViewID_OVR == 0u) ? LEFT_VIEW_MATRIX : RIGHT_VIEW_MATRIX,
+    MODEL_MATRIX);
+}
+`;
+
+const VERTEX_SHADER_DEPTH_ENTRY = `
+uniform mat4 PROJECTION_MATRIX, VIEW_MATRIX, MODEL_MATRIX;
+out vec4 vWorldPosition;
+
+const mat4 identity = mat4(1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1);
+void main() {
+  vWorldPosition = vertex_main(identity, identity, MODEL_MATRIX);
+  gl_Position = vertex_main(PROJECTION_MATRIX, VIEW_MATRIX, MODEL_MATRIX);
+}
+`;
+
+const FRAGMENT_SHADER_DEPTH_COMMON = `
+precision highp float;
+precision highp sampler2DArray;
+uniform sampler2DArray depthColor;
+out vec4 color;
+in vec4 vWorldPosition;
+uniform mat4 LEFT_DEPTH_PROJECTION_MATRIX, LEFT_DEPTH_VIEW_MATRIX, RIGHT_DEPTH_PROJECTION_MATRIX, RIGHT_DEPTH_VIEW_MATRIX;
+
+float Depth_GetCameraDepthInMillimeters(const sampler2DArray depthTexture,
+  const vec2 depthUv) {
+  return texture(depthColor, vec3(depthUv.x, depthUv.y, VIEW_ID)).r * 1000.0;
+}
+
+float Depth_GetOcclusion(const sampler2DArray depthTexture, const vec2 depthUv, float assetDepthMm) {
+  float depthMm = Depth_GetCameraDepthInMillimeters(depthTexture, depthUv);
+
+  // Instead of a hard z-buffer test, allow the asset to fade into the
+  // background along a 2 * kDepthTolerancePerMm * assetDepthMm
+  // range centered on the background depth.
+  const float kDepthTolerancePerMm = 0.01;
+  return clamp(1.0 -
+    0.5 * (depthMm - assetDepthMm) /
+        (kDepthTolerancePerMm * assetDepthMm) +
+    0.5, 0.0, 1.0);
+}
+
+float Depth_GetBlurredOcclusionAroundUV(const sampler2DArray depthTexture, const vec2 uv, float assetDepthMm) {
```

Comment:
Explicitly going to defer to Brandon on this whole function.

Reply:
I got this code from a google sample

Reply:
Got it. Not seeing the `/SideBySideMultiplier` in the Google sample I found that matched this, and it looks like you've commented out `/** u_DepthAspectRatio*/`

As long as those are intentional, then this LGTM.